### PR TITLE
ENHO: workaround dump during serialization

### PR DESCRIPTION
**Pull request by larshp**
During serialization of ENHO, the SAP standard code might dump in method `get_include_bound()`, this change introduces a workaround to avoid this short dump

### CODE REVIEW

File: src/objects/enh/zcl_abapgit_object_enho_hook.clas.abap [hunk 1]
```diff
@@ -254,7 +255,23 @@ CLASS zcl_abapgit_object_enho_hook IMPLEMENTATION.
         main_type = ls_original_object-org_main_type
         main_name = ls_original_object-org_main_name
         program   = ls_original_object-programname ).
-    ls_original_object-include_bound = lo_hook_impl->get_include_bound( ).
+
+* dont call method lo_hook_impl->get_include_bound( ), it might dump
+* if the PROG does not exists
+    IF ls_original_object-org_main_type = 'PROG' OR ls_original_object-org_main_type = 'REPS'.
+      CALL FUNCTION 'READ_TRDIR'
```
Comment (mbtools) [inline src/objects/enh/zcl_abapgit_object_enho_hook.clas.abap:262]: Maybe zcl_abapgit_sap_report->read_progdir ?

Comment (larshp) [reply src/objects/enh/zcl_abapgit_object_enho_hook.clas.abap]: good catch, code updated

**Review by mbtools — APPROVED** [review]
👍

**Comment by pokrakam** [comment]
Curiosity question: I thought we do these in lowercase, and abaplint by default complains about it so why the change?
image
If I change a class name to upper I get an error on my system:
image

**Comment by larshp** [comment]
> Curiosity question: I thought we do these in lowercase, and abaplint by default complains about it so why the change?

SE80 changes it, so its bit dodgy

https://github.com/abapGit/abapGit/blob/main/abaplint.json#L243 sets
`"ignoreLowerClassImplmentationStatement": true,` so the linter should not care about this specific statement being upper or lower

**Pull request merged**